fresque/gitview.py

```diff
@@ -0,0 +1,221 @@
+# -*- coding: utf-8 -*-
+
+'''
+package for the fresque flask application
+'''
+
+from __future__ import absolute_import, unicode_literals, print_function
+
+import os
+import flask
+import pygit2
+import fresque
+
+import kitchen.text.converters as ktc
+
+from sqlalchemy.orm import aliased
+from pygments import highlight
+from pygments.formatters import HtmlFormatter
+from pygments.lexers import guess_lexer_for_filename
+from pygments.lexers.special import TextLexer
+from pygments.util import ClassNotFound
+
+import mimetypes
+import chardet
+
+from fresque import APP
+from fresque.lib.git import Repository
+
+
+def get_repo_by_name(name):
+    path = os.path.join(
+        APP.config['GIT_DIRECTORY_PATH'], flask.g.fas_user.username)
+    try:
+        repo_obj = Repository(os.path.join(path, name))
+    except IOError:
+        return "No such repo", 404
```

Comment:
I wonder between try/except and using ``os.path.exists()``, maybe the try/except is safer as it ensure it's valid git repository.